src/segment-loader.js

```diff
@@ -407,19 +407,32 @@ export const shouldWaitForTimelineChange = ({
   return false;
 };
 
-export const mediaDuration = (audioTimingInfo, videoTimingInfo) => {
-  const audioDuration =
-    audioTimingInfo &&
-    typeof audioTimingInfo.start === 'number' &&
-    typeof audioTimingInfo.end === 'number' ?
-      audioTimingInfo.end - audioTimingInfo.start : 0;
-  const videoDuration =
-    videoTimingInfo &&
-    typeof videoTimingInfo.start === 'number' &&
-    typeof videoTimingInfo.end === 'number' ?
-      videoTimingInfo.end - videoTimingInfo.start : 0;
-
-  return Math.max(audioDuration, videoDuration);
+export const mediaDuration = (timingInfos) => {
+  let maxDuration = 0;
+
+  ['video', 'audio'].forEach(function(type) {
+    const typeTimingInfo = timingInfos[`${type}TimingInfo`];
+
+    if (!typeTimingInfo) {
+      return;
+    }
+    const {start, end} = typeTimingInfo;
+    let duration;
+
+    // eslint-disable-next-line
+    if (typeof start === 'bigint' || typeof end === 'bigint') {
+      duration = window.BigInt(end) - window.BigInt(start);
```

Comment:
each item needs to be wrapped in `BigInt()` to make sure that JS runs it in BigInt mode and doesn't try to run it in regular int mode?

Reply:
That is correct. JavaScript throws an error if you try to do any math between `BigInt` and a normal `Number`. Once you have one `BigInt` you have to stick with it.